libavformat/panr.c

```diff
@@ -188,38 +203,56 @@ static int read_header(AVFormatContext * format_ctx)
         // to try to get it right
         if (avst->codecpar->bit_rate < 20000)
         {
-                avst->codecpar->bit_rate *= 1000;
+            av_log(format_ctx, AV_LOG_TRACE, "Parsed bitrate was too low, multiplying up by 1000\n");
+            avst->codecpar->bit_rate *= 1000;
         }
     }
     else if (memcmp(&demux_ctx->file_header.majortype, &PANR_MEDIATYPE_Audio, sizeof(GUID)) == 0)
     {
         if (memcmp(&demux_ctx->file_header.formattype, &PANR_FORMAT_WaveFormatEx, sizeof(GUID)) != 0)
         {
             ret = AVERROR_INVALIDDATA;
+            av_log(format_ctx, AV_LOG_ERROR, "Detected audio format header type was not WaveFormatEx, and is thus not supported \n");
             goto Cleanup;
         }
 
         wave_format = (WAVEFORMATEX*)demux_ctx->format_block;
         avst->codecpar->codec_type = AVMEDIA_TYPE_AUDIO;
         avst->codecpar->codec_id = AV_CODEC_ID_AAC;
-        avst->codecpar->codec_tag = 0;
+        avst->codecpar->codec_tag = MKTAG('m', 'p', '4', 'a');
 
+        avst->codecpar->profile = FF_PROFILE_AAC_LOW;
         avst->codecpar->channels = wave_format->nChannels;
-        avst->codecpar->channel_layout = wave_format->nChannels == 2? AV_CH_LAYOUT_STEREO : AV_CH_LAYOUT_MONO;
+        avst->codecpar->channel_layout = wave_format->nChannels == 2 ? AV_CH_LAYOUT_STEREO : AV_CH_LAYOUT_MONO;
         avst->codecpar->sample_rate = wave_format->nSamplesPerSec;
         avst->codecpar->block_align = wave_format->nBlockAlign;
-        avst->codecpar->frame_size = wave_format->wBitsPerSample;
+
+        // prepare some data to emit in the esds audio specific config via side_data
+        // see https://wiki.multimedia.cx/index.php?title=MPEG-4_Audio
+        // (more context: https://stackoverflow.com/questions/3987850/mp4-atom-how-to-discriminate-the-audio-codec-is-it-aac-or-mp3)
+        demux_ctx->audio_object_type = AOT_AAC_LC;
+        demux_ctx->audio_sampling_index = 0;
+        for (int i = 0; i < sizeof(avpriv_mpeg4audio_sample_rates) / sizeof(avpriv_mpeg4audio_sample_rates[0]); i++)
+        {
+            if (avst->codecpar->sample_rate == avpriv_mpeg4audio_sample_rates[i])
+            {
+                demux_ctx->audio_sampling_index = i;
```

Comment:
Maybe you should `break;` out here.  Also, what happens if we never enter this part?  Is that something that you can detect and then log?

Reply:
I wonder if this is just iterating the values in the "Sampling Frequencies" section of [the wiki](https://wiki.multimedia.cx/index.php?title=MPEG-4_Audio), in which case it seems like the escape hatch is to use an escape value (15) to write the sampling frequency explicitly.

Reply:
Realistically panras are always 44.1 or 48 because we resample to those. Adding the ability to use 15 adds a decent amount of complexity sadly that we'd never use, so I'll opt to default to 44.1 if anything goes wrong with detection (realistically shouldn't cause an issue in the majority of cases I suspect). Will also log in that case...

Reply:
Sounds good.